staging/src/k8s.io/code-generator/cmd/validation-gen/validators/subfield.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validators
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/gengo/v2/parser/tags"
+	"k8s.io/gengo/v2/types"
+)
+
+const (
+	subfieldTag = "k8s:subfield2"
+)
+
+func init() {
+	RegisterTagValidator(&subfieldTagValidator{})
+}
+
+type subfieldTagValidator struct {
+	validator Validator
+}
+
+func (stv *subfieldTagValidator) Init(cfg Config) {
+	stv.validator = cfg.Validator
+}
+
+func (subfieldTagValidator) TagName() string {
+	return subfieldTag
+}
+
+var subfieldTagValidScopes = sets.New(ScopeAny)
+
+func (subfieldTagValidator) ValidScopes() sets.Set[Scope] {
+	return subfieldTagValidScopes
+}
+
+var (
+	validateSubfield = types.Name{Package: libValidationPkg, Name: "Subfield"}
+)
+
+func (stv subfieldTagValidator) GetValidations(context Context, args []string, payload string) (Validations, error) {
+	t := context.Type
+	if t.Kind == types.Alias {
+		t = t.Underlying
+	}
+	if t.Kind != types.Struct {
+		return Validations{}, fmt.Errorf("can only be used on struct types")
+	}
+	if len(args) != 1 {
+		return Validations{}, fmt.Errorf("requires exactly one arg")
+	}
+	subname := args[0]
+	submemb := getMemberByJSON(t, subname)
+	if submemb == nil {
+		return Validations{}, fmt.Errorf("no field for json name %q", subname)
+	}
+
+	result := Validations{}
+
+	fakeComments := []string{payload}
+	subContext := Context{
+		Scope:  ScopeField,
+		Type:   submemb.Type,
+		Parent: t, //FIXME: t or context.Type?
```

Comment:
+1 for `t`, I believe using the resolved underlying types here rather than aliases is preferred IIUC.